Source/Scene/ModelOutlineLoader.js

```diff
@@ -0,0 +1,703 @@
+import binarySearch from "../Core/binarySearch.js";
+import defined from "../Core/defined.js";
+import PixelFormat from "../Core/PixelFormat.js";
+import ContextLimits from "../Renderer/ContextLimits.js";
+import Sampler from "../Renderer/Sampler.js";
+import Texture from "../Renderer/Texture.js";
+import TextureMagnificationFilter from "../Renderer/TextureMagnificationFilter.js";
+import TextureMinificationFilter from "../Renderer/TextureMinificationFilter.js";
+import TextureWrap from "../Renderer/TextureWrap.js";
+import ForEach from "../ThirdParty/GltfPipeline/ForEach.js";
+
+/**
+ * Creates face outlines for glTF primitives with the `CESIUM_primitive_outline` extension.
+ * @private
+ */
+function ModelOutlineLoader() {}
+
+/**
+ * Returns true if the model uses or requires CESIUM_primitive_outline.
+ * @private
+ */
+ModelOutlineLoader.hasExtension = function (model) {
+  return (
+    defined(model.extensionsRequired.CESIUM_primitive_outline) ||
+    defined(model.extensionsUsed.CESIUM_primitive_outline)
+  );
+};
+
+/**
+ * Arranges to outline any primitives with the CESIUM_primitive_outline extension.
+ * It is expected that all buffer data is loaded and available in
+ * `extras._pipeline.source` before this function is called, and that vertex
+ * and index WebGL buffers are not yet created.
+ * @private
+ */
+ModelOutlineLoader.outlinePrimitives = function (model) {
+  if (!ModelOutlineLoader.hasExtension(model)) {
+    return;
+  }
+
+  var gltf = model.gltf;
+
+  // Assumption: A single bufferView contains a single zero-indexed range of vertices.
+  // No trickery with using accessor byteOffsets to store multiple zero-based ranges of
+  // vertices in a single bufferView. Use separate bufferViews for that, you monster.
```

Comment:
Oh yeah that's definitely still common practice, and it's supported here. In fact the main dataset I've been using to test this has interleaved vertex attributes.

That comment is not referring to interleaved attributes (e.g. pos1, normal1, uv1, pos2, normal2, uv2, etc.) but to dodgily packing two essentially independent `VertexBuffer`s into a single bufferView by using a separate set of accessors with _large_ byteOffsets. Certainly that's not common practice, and I'm not sure it would even work in CesiumJS, though I can't see anything in the spec that would forbid it. I'll tweak the comment to make this more clear.